pcdsdevices/type_hints.py

```diff
@@ -29,13 +29,19 @@ def __call__(**kwargs) -> None:
 SignalToValue = Dict[ophyd.Signal, OphydDataType]
 
 
-class MdsCalculateFunction(Protocol):
+@runtime_checkable
+class MdsOnGetFunction(Protocol):
```

Comment:
I guess I'm just skeptical/inexperienced: is this as magical as it looks? Define a `Protocol` subclass, slap on a decorator, and now it can be used in an `isinstance` check for an arbitrary callable? Let python take the wheel?

What exactly is the `isinstance` check looking for when we compare a function to these protocols? Call signature? Argument count? Annotations?

Reply:
You were right to be skeptical. The removal of the signature check is wrong and should be covered by the test suite.

https://docs.python.org/3/library/typing.html#typing.runtime_checkable

I would either re-add the original checks or a utility function for "check signature against Protocol".

Reply:
Resolved in c1b8b97db1